operate/operate_types.py

```diff
@@ -203,6 +203,23 @@ class ConfigurationTemplate(TypedDict):
     fallback_chain_params: t.Optional[t.Dict]
 
 
+class ServiceEnvProvisionType(enum.Enum):
+    """Service environment variable provision type."""
+
+    FIXED = enum.auto()
+    USER = enum.auto()
+
+
+class ServiceEnvVariable(TypedDict):
+    """Service environment variable template."""
+
+    name: str
+    env_variable_name: str
+    description: str
+    value: t.Union[str, int, bool, None]
```

Comment:
This is what worries me more about. Although there are several types of env variables in shell, when setting them up in python then have to be always strings AFAIK.
`os.environ["MY_VAR"] = "1"`

For this reason, I would only consider strings.